src/coreclr/src/vm/methodtablebuilder.cpp

```diff
@@ -11282,30 +11301,29 @@ BOOL MethodTableBuilder::NeedsAlignedBaseOffset()
             return FALSE;
     }
 
-    if (pParentMT->GetModule() == GetModule())
-    {
-        if (!pParentMT->GetClass()->HasLayoutDependsOnOtherModules())
-            return FALSE;
-    }
-    else
-    {
+    bool treatAsIfInSameModule = pParentMT->GetModule() == GetModule();
```

Comment:
Exactly the same logic is now duplicated here and in the MethodTableBuilder::CheckLayoutDependsOnOtherModules. So it seems that this whole block can be replaced by
```
        return pParentMT->GetClass()->HasLayoutDependsOnOtherModules();
```